### PR TITLE
Redhat_OS_update_roles

### PR DESCRIPTION
**Pull request by ioigoume**
httpd Role update

### CODE REVIEW

File: roles/rciam-probes/tasks/install-RedHat.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  yum_repository:
+    name: "{{ item.name }}"
+    description: "{{ item.description }}"
+    file: "{{ item.file }}"
+    baseurl: "{{ item.baseurl }}"
+    gpgcheck: "{{ item.gpgcheck}}"
+    enabled: "{{ item.enabled }}"
+  loop: "{{ repos | default([])}}"
+  become: yes
+  notify: yum Update
```
Comment (NicolasLiampotis) [inline roles/rciam-probes/tasks/install-RedHat.yml:13]: Is this necessary? See comment about yum update handler below

File: roles/rciam-probes/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+
+- name: Clean cache
+  command: "{{ clean_cache_cmd }}"
+  listen: "yum Update"
+  become: yes
+
+- name: yum Update (RedHat)
+  yum:
+    name: '*'
```
Comment (NicolasLiampotis) [inline roles/rciam-probes/handlers/main.yml:10]: Upgrading all OS packages can cause unpredictable effects. Can we avoid this?

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: roles/apache/tasks/configure-Debian.yml [hunk 3]
```diff
@@ -18,6 +18,17 @@
...
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: root
+    group: root
+  loop:
+    - "{{ apache_conf_path }}/sites-available"
+    - "{{ apache_conf_path }}/sites-enabled"
+  become: yes
+
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/apache/tasks/configure-Debian.yml]: Done

File: roles/apache/tasks/configure-RedHat.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    dest: "{{ apache_conf_path }}/conf.d/security.conf"
+    owner: root
+    group: root
+    mode: 0644
+    backup: yes
+  become: yes
+  notify: restart apache
+
+# Configure Virtual Host SSL certificates
+- name: Iclude vhost SSL certificates configuration (RedHat)
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/apache/tasks/configure-RedHat.yml]: done

File: roles/apache/tasks/configure-RedHat.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+- name: Enable firewalld http(s) services (RedHat)
+  firewalld:
+    service: "{{ item }}"
+    permanent: true
+    state: enabled
+  loop:
+    - http
+    - https
+  become: true
+  notify: reload firewall
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/apache/tasks/configure-RedHat.yml]: done

File: roles/apache/tasks/configure-RedHat.yml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+- name: Enable Apache vhosts configuration (RedHat)
+  file:
+    src: "{{ apache_conf_path }}/sites-available/{{ apache_vhosts_filename }}"
+    dest: "{{ apache_conf_path }}/sites-enabled/{{ apache_vhosts_filename }}"
+    state: link
+  become: yes
+  notify: restart apache
+  when: apache_create_vhosts
+
+- name: Disable welcome screen
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/apache/tasks/configure-RedHat.yml]: Done

File: roles/apache/tasks/install-RedHat.yml [hunk 7]
```diff
@@ -1,8 +1,15 @@
...
   yum: 
-    name: httpd
-    state: present 
+    name: "{{ item }}"
+    state: latest
     update_cache: yes
   become: yes
+  loop:
+    - httpd
+    - mod_ssl
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/apache/tasks/install-RedHat.yml]: Done. Created two separate tasks. The one that installs httpd and a second one which install mod_ssl if `apache_vhosts_ssl` is not empty

File: roles/rciam-probes/handlers/main.yml (hunk 2 above)
Comment (ioigoume) [reply roles/rciam-probes/handlers/main.yml]: This is now a task which only fire with the tag upgrade. We need this when deploying for the very first time.

File: roles/php/tasks/install-RedHat.yml [hunk 8]
```diff
@@ -1,11 +1,24 @@
...
+    name: "{{ item }}"
+    state: present
+    update_cache: yes
+  loop:
+    - http://rpms.remirepo.net/enterprise/remi-release-{{ centos_version }}.rpm
+  become: true
+
+- name: Enable Remi repository (RedHat)
+  command: "yum-config-manager --enable remi-php{{ php_version }} && yum clean all && yum update"
+  become: true
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/php/tasks/install-RedHat.yml]: - Yum repository does not work really well with custom packages that include multiple repositories. Yet, i refactored the code following a more clear approach.
- Used `php_alt_repo` flag as dictated

File: roles/rciam-probes/tasks/install-RedHat.yml (hunk 1 above)
Comment (ioigoume) [reply roles/rciam-probes/tasks/install-RedHat.yml]: removed

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
Regarding the apache role, we need to avoid introducing vars specifically for Centos/RedHat. For example, why do we need to the apache_ssl when there is already apache_ssl_vhosts?

Please have a look at https://github.com/geerlingguy/ansible-role-apache/blob/master/templates/vhosts.conf.j2 which is supposed to support both Debian and RedHat systems. It might be an option to switch to this role and deal with the ssl key/cert copy as a different task/role.

File: roles/rciam-probes/vars/CentOS-8.yml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (NicolasLiampotis) [inline roles/rciam-probes/vars/CentOS-8.yml:1]: Should this be renamed to RedHat-8.yml for consistency with RedHat.yml in other roles?

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
Check if we can use RedHat-X.yml instead of CentOS-X.yml. Other than that LGTM

File: roles/rciam-probes/vars/CentOS-7.yml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+---
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/rciam-probes/vars/CentOS-7.yml]: Done

File: roles/rciam-probes/vars/CentOS-8.yml (hunk 9 above)
Comment (ioigoume) [reply roles/rciam-probes/vars/CentOS-8.yml]: Done

File: roles/php/tasks/install-RedHat.yml [hunk 11]
```diff
@@ -1,19 +1,39 @@
...
+    key: "{{ item }}"
+    state: present
+  loop: "{{ php_yum_keys }}"
+  when: php_alt_repo and php_yum_repositories is defined and php_yum_repositories|length
+  become: true
+
 - name: Ensure PHP packages are installed (RedHat)
   yum:
-    name: "{{ php_packages }}"
+    enablerepo: "remi,remi-safe,remi-php{{ php_version }}"
```
Comment (NicolasLiampotis) [inline roles/php/tasks/install-RedHat.yml:21]: isn't remi required only when php_alt_repo is true?

File: roles/php/tasks/install-RedHat.yml [hunk 12]
```diff
@@ -1,19 +1,39 @@
...
+    name: "{{ item }}"
     state: present
     update_cache: yes
   become: true
+  loop: "{{ php_default_packages }}"
```
Comment (NicolasLiampotis) [inline roles/php/tasks/install-RedHat.yml:26]: this should `php_packages`. see task in `tasks/main.yml`:
```
- name: Define PHP packages
  set_fact:
    php_packages: "{{ php_default_packages | list }}"
  when: php_packages is not defined
```

File: roles/php/tasks/install-RedHat.yml [hunk 13]
```diff
@@ -1,19 +1,39 @@
...
   notify:
     - restart webserver
 
 - name: Ensure PHP extensions are installed (RedHat)
   yum:
+    enablerepo: "remi,remi-safe,remi-php{{ php_version }}"
```
Comment (NicolasLiampotis) [inline roles/php/tasks/install-RedHat.yml:32]: isn't remi required only when `php_alt_repo` is true?

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
1. Use `php_packages` instead of `php_default_packages`.
2. Check how remi can be enabled based on `php_alt_repo`

File: roles/php/tasks/install-RedHat.yml (hunk 11 above)
Comment (ioigoume) [reply roles/php/tasks/install-RedHat.yml:21]: Done

File: roles/php/tasks/install-RedHat.yml (hunk 12 above)
Comment (ioigoume) [reply roles/php/tasks/install-RedHat.yml]: Done

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**